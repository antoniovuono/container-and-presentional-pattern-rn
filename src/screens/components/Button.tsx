import {
  TouchableOpacity,
  Text,
  StyleSheet,
  TouchableOpacityProps,
  ActivityIndicator,
} from 'react-native';

type ButtonProps = TouchableOpacityProps & {
  title: string;
  isLoading: boolean;
};

export function Button({ title, isLoading, ...rest }: ButtonProps) {
  return (
    <TouchableOpacity style={styles.button} {...rest}>
      {isLoading ? (
        <ActivityIndicator
          testID="loading-indicator"
          size="small"
          color="white"
        />
      ) : (
        <Text style={styles.buttonText}>{title}</Text>
      )}
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#00B37E',
    padding: 15,
    borderRadius: 8,
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
  },
  buttonText: {
    color: '#fff',
    fontSize: 12,
    fontWeight: 'bold',
  },
});
