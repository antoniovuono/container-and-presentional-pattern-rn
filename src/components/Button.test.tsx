import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { Button } from './Button';

const renderButton = (isLoading: boolean) => {
  return render(<Button title={'Entrar'} isLoading={isLoading} />);
};

describe('Button', () => {
  it('should render button correctly', () => {
    const tree = renderButton(false).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be able to render a loading indicator if isLoading is true', () => {
    renderButton(true);

    const loadingIndicatorElement = screen.getByTestId('loading-indicator');

    expect(loadingIndicatorElement).toBeDefined();
  });
});
