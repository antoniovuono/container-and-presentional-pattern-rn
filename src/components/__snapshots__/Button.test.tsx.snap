// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00B37E",
      "borderRadius": 8,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 12,
        "fontWeight": "bold",
      }
    }
  >
    Entrar
  </Text>
</View>
`;
